### PR TITLE
Documentation on save, load, and resume experiments

### PR DESCRIPTION
**Pull request by irisliucy**
This PR covers #1535 &  #1537

### CODE REVIEW

File: docs/user/pretrained_network_experiment.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+This tutorial will walk you through starting an experiment with a pre-trained network using two of the most common deep learning frameworks, Tensorflow and Pytorch.
+
+## Set up the environment
+
+First, we will start by setting up the environment for the experiment. We will train a `VPG algorithm` as an example.
+
+We will simply setup the environment by initializing the garage environment `GarageEnv` and the `LocalRunner` instance.
+
+```python
+def vpg_pendulum(ctxt=None, seed=1):
```
Comment (ryanjulian) [inline docs/user/pretrained_network_experiment.md:14]: maybe use PPO or TROP, which are more typical algorithms (VPG is pretty bad performance)

File: docs/user/pretrained_network_experiment.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    set_seed(seed)
+    env = GarageEnv(env_name='InvertedDoublePendulum-v2')
+
+    runner = LocalRunner(ctxt)
+
+    ....
+```
+
+## Load the pre-trained model
```
Comment (ryanjulian) [inline docs/user/pretrained_network_experiment.md:23]: we don't actually use these mechanisms for saving models, though maybe we should. instead, we pickle the entire model object (`Policy`, `QFunction`, etc.)

check out the examples folder

**Comment by ryanjulian** [comment]
I think you need to update `docs/index.rst` so this page appears in the table of contents.

**Comment by ryanjulian** [comment]
Don't forget to add an acknowledgement for yourself at the bottom of the page!  See https://github.com/rlworkgroup/garage/pull/1638 for an example.

**Comment by irisliucy** [comment]
@Mergifyio rebase

File: docs/user/pretrained_network_experiment.md [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+
+In general starting a new experiment with a pre-trained
+network follow the same few steps:
+
+- Initialize the `Snapshotter` instance
+- Restore the pre-trained model
+- Define for the parameters we want to update during training
+- Run the training step with `resume`
+
+In this tutorial, we will walk you through how to use
```
Comment (ryanjulian) [inline docs/user/pretrained_network_experiment.md:11]: We have a separate Tutorials section, so please don't call this a tutorial :)

File: docs/user/pretrained_network_experiment.md [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+a pre-trained network to start a new experiment with
+`TRPO algorithm` in garage. In practice, garage allows
```
Comment (ryanjulian) [inline docs/user/pretrained_network_experiment.md:13]: `TRPO`

**Comment by ryanjulian** [comment]
I think this almost covers #1537 completely -- can you amend it to be that page?

We can cover the original issue in the next RC. I think I did a poor job of explaining my intention for this page -- my apologies.

File: docs/user/pretrained_network_experiment.md [hunk 5]
```diff
@@ -0,0 +1,237 @@
+# Use a pre-trained network to start a new experiment
+
+In general starting a new experiment with a pre-trained
```
Comment (yeukfu) [inline docs/user/pretrained_network_experiment.md:3]: 'In general, starting a new experiment with a pre-trained'

File: docs/user/pretrained_network_experiment.md [hunk 6]
```diff
@@ -0,0 +1,237 @@
...
+2020-06-26 13:42:16 | epoch #18 | Obtaining samples for iteration 18...
+2020-06-26 13:42:17 | epoch #18 | Logging diagnostics...
+2020-06-26 13:42:17 | epoch #18 | Optimizing policy...
+2020-06-26 13:42:17 | epoch #18 | Computing loss before
+2020-06-26 13:42:17 | epoch #18 | Computing KL before
+2020-06-26 13:42:17 | epoch #18 | Optimizing
+2020-06-26 13:42:17 | epoch #18 | Start CG optimization:
+#parameters: 1282, #inputs: 52, #subsample_inputs: 52
+2020-06-26 13:42:17 | epoch #18 | computing loss before
+2020-06-26 13:42:17 | epoch #18 | computing gradient
```
Comment (yeukfu) [inline docs/user/pretrained_network_experiment.md:151]: Maybe showing one epoch output is enough?

**Comment by irisliucy** [comment]
The doc is now updated and move to cover section #1537 instead.

File: docs/user/save_load_resume_exp.md [hunk 7]
```diff
@@ -0,0 +1,309 @@
+# Save, load and resume experiments
+
+This document provides solutions to a variety of use cases
+regarding saving, loading and resuming of Garage experiments.
+
+**Contents**:
+
+- [Local_runner/ Tf_local_runner for Garage experiment](#localrunner-localtfrunner-for-garage-experiment)
```
Comment (ryanjulian) [inline docs/user/save_load_resume_exp.md:8]: Please use the code names `LocalRunner` and `LocalTFRunner`

File: docs/user/save_load_resume_exp.md [hunk 8]
```diff
@@ -0,0 +1,309 @@
...
+- [Saving & Training Models in an experiment](#saving--training-models-in-an-experiment)
+- [Loading Models & Resuming an experiment](#loading-models--resuming-an-experiment)
+
+```python
+import garage
+from garage.experiment import LocalRunner, LocalTFRunner
+from garage.experiment.deterministic import set_seed
+```
+
+## `Local_runner`/ `Local_tf_runner` for Garage experiment
```
Comment (ryanjulian) [inline docs/user/save_load_resume_exp.md:18]: `LocalRunner` and `LocalTFRunner`

File: docs/user/save_load_resume_exp.md [hunk 9]
```diff
@@ -0,0 +1,309 @@
...
+policy/KL                                     0.00595805
+policy/KLBefore                               0
+policy/LossAfter                             -0.00285657
+policy/LossBefore                            -2.5089e-07
+policy/Perplexity                             1.69811
+policy/dLoss                                  0.00285632
+---------------------------------------  ---------------
+```
+
+You may find the full example at `examples/jupyter/pre-trained_trpo_gym_tf_cartpole.ipynb`.
```
Comment (ryanjulian) [inline docs/user/save_load_resume_exp.md:306]: Can you point to a non-Jupyter example? We don't really maintain this very well right now.

Comment (irisliucy) [reply docs/user/save_load_resume_exp.md]: Removed `examples/jupyter/pre-trained_trpo_gym_tf_cartpole.ipynb`. Instead, add example `examples/tf/trpo_gym_tf_cartpole_pretrained.py`

**Comment by irisliucy** [comment]
@Mergifyio rebase

**Comment by ryanjulian** [comment]
LGTM

File: docs/user/save_load_resume_exp.md [hunk 10]
```diff
@@ -0,0 +1,211 @@
...
+- Initialize the `LocalRunner`/ `LocalTFRunner` instance
+- Restore the algorithm and experiment setup with `LocalRunner.restore`
+- Define for the parameters we want to update during training
+- Run the training step with `LocalRunner.resume`
+
+```Python
+# to resume immediately.
+runner = LocalRunner()
+runner.restore(resume_from_dir)
+runner.resum
```
Comment (yeukfu) [inline docs/user/save_load_resume_exp.md:91]: `runner.resume`

File: docs/user/save_load_resume_exp.md [hunk 11]
```diff
@@ -0,0 +1,211 @@
...
+policy/LossAfter                              0.00548463
+policy/LossBefore                             0.0165238
+policy/Perplexity                             1.82338
+policy/dLoss                                  0.0110392
+....
+
+---------------------------------------  ---------------
+```
+
+You may find the full example at `examples/tf/trpo_gym_tf_cartpole_pretrained.py`.
```
Comment (yeukfu) [inline docs/user/save_load_resume_exp.md:208]: Personally, it will be better if this can link to the file.

**Comment by irisliucy** [comment]
@Mergifyio rebase

File: docs/user/save_load_resume_exp.md [hunk 12]
```diff
@@ -0,0 +1,211 @@
...
+- Initialize the `LocalRunner`/ `LocalTFRunner` instance
+- Restore the algorithm and experiment setup with `LocalRunner.restore`
+- Define for the parameters we want to update during training
+- Run the training step with `LocalRunner.resume`
+
+```Python
+# to resume immediately.
+runner = LocalRunner()
+runner.restore(resume_from_dir)
+runner.resume
```
Comment (zequnyu) [inline docs/user/save_load_resume_exp.md:91]: bracket?

Comment (irisliucy) [reply docs/user/save_load_resume_exp.md:91]: Thanks for the spotting. Opened #1692 to fix that.
